### PR TITLE
docs: add error handling ADR

### PR DESCRIPTION
**Pull request by kamilbaczek**

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+== Consequences
+- The Result type approach will simplify the domain logic.
+- The error flow will be explicit and easier to follow flow.
+- Performance will be improved by reducing the impact of exception handling in negative scenarios.
+- Team will need to adhere to the defined guidelines for using the Result type approach.
+- The team will need to allocate time for refactoring the existing code to adopt the new approach.
+
+== Resources
+link:https://youssefsellami.com/exceptions_vs_result_object/[Benchmark]
```
Comment (meaboutsoftware) [inline Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc:46]: Move this to the problem description and mention that this is one of the main reasons of this architecture decision record

File: Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Date: 2024-07-23
+
+== Problem
+
+On current project state Contracts module bussiness logic is getting more complex and it is hard to control flow of errors in the module. 
+
+The current error handling approach is based on exceptions, which makes it difficult to follow the appliation flow and maintain the codebase. The team is considering adopting a new approach to handle errors in the Contracts module.
+
+== Options
```
Comment (meaboutsoftware) [inline Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc:11]: Options can be a part of Problem (===). This way we do not change the structure of ADRs, therefore no one will be mislead by completely different ADR structure in comparison to others

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

File: Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc (hunk 2 above)
Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc:11]: Correct :D I will change it.

File: Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc (hunk 1 above)
Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Docs/ArchitectureDecisionLog/0024-contracts-error-handling.adoc:46]: Ok

**Review by meaboutsoftware — APPROVED** [review]

**Pull request merged**